proposals/declarative-theme-media-breakpoints.md

```diff
@@ -0,0 +1,67 @@
+# Declarative Theme Media Breakpoints in `etc/view.xml`
+
+## Context
+
+Currently we declare theme breakpoints through LESS variables, notably `@screen__m` declares the breakpoint at which the design should change to a mobile design.
+
+Page Builder requires more dynamic generation of styles due to the dynamic nature of how the content is created. We have background image functionality which requires the generation of CSS to provide a high performant solution, due to this we need to be able to determine the current themes mobile breakpoint.
+
+## Implementation Strategy
+
+Currently themes implement a `etc/view.xml` configuration file which contains module specific configuration.
+
+For instance the file for both `luma` and `blank` contains the following nodes:
+
+```xml
+<vars module="Magento_Catalog">
+	...
+    <var name="breakpoints">
+        <var name="mobile">
+            <var name="conditions">
+                <var name="max-width">767px</var>
+            </var>
+            <var name="options">
+                <var name="options">
+                    <var name="nav">dots</var>
+                </var>
+            </var>
+        </var>
+    </var>
+    ...
+</vars>
+```
+
+We would look to unify the breakpoint configuration for all themes under a single node and not specific for a particular module. 
+
+### Long Term Solution
+
+In the long term we'd like to be able to support population of LESS variables from the themes XML configuration. This allows for the greater Magento system to have an understanding of when various different content should be displayed, this could be represented in the themes XML as follows:
+
+```
+<theme_variables>
+    <var name="screen__m">767px</var>
+    <var name="default_font_size">14px</var>
+</theme_variables>
+```
+
+These variables names would be converted over to `@screen__m` and `@default_font_size` respectively.
+
+### Short Term Page Builder Specific Solution
+
+The long term solution has a large requirement of resource and careful design to ensure it's implementation fits it's purpose correctly.
+
+Page Builder requires an understanding of break points now ,_similar to how Magento_Catalog does currently_, due to this we can introduce new configuration specific for our module, within the `blank` and `luma` themes` as follows:
+
+```
+<vars module="Magento_PageBuilder">
```

Comment:
We can have this configuration in the PageBuilder module and we allow to override it from the theme.

For the future we should consider generic solution that would allow us to pass different variables from configuration to LESS(option 1). We need to give good thought to option 1 configuration and implementation as well as receive feedback from community. Option 2 seems like good solution for now, that we can deprecate when we implement option 1.

Reply:
It's not currently implemented for the specific issue @davemacaulay is trying to solve.

Reply:
After talking in Slack I think we are on the same page. I misunderstood your comment.

Reply:
I think if we can solve our problem using Option 2 it is better to proceed with it.